topic_tools/src/tool_base_node.cpp

```diff
@@ -39,17 +39,26 @@ void ToolBaseNode::make_subscribe_unsubscribe_decisions()
       std::scoped_lock lock(pub_mutex_);
       pub_ = this->create_generic_publisher(output_topic_, *topic_type_, *qos_profile_);
     }
-
     // at this point it is certain that our publisher exists
+    
+    // If lazy, only subscribe to input_topic if there is at least one subscriber on the output_topic
     if (!lazy_ ||
       pub_->get_subscription_count() + pub_->get_intra_process_subscription_count() > 0)
     {
       if (!sub_) {
         sub_ = this->create_generic_subscription(
           input_topic_, *topic_type_, *qos_profile_,
           std::bind(&ToolBaseNode::process_message, this, std::placeholders::_1));
+      } 
+      else {
+        // subscription exists already but needs changing if input_topic_ changes
```

Comment:
This is going to wait one more `discovery_period_` before creating the new subscription, right? It would be better if it did this immediately the first time we notice the change, like

```
if (sub_ && sub_->topic != input_topic_) {
  sub_.reset();
}
if (!sub_) {
  // create sub
}
```

Reply:
I thought about it but I thought it was the previous behaviour and didn't want to change it much. I will do the change. Thanks!